common/docker-entrypoint.d/00-check-for-required-env.sh

```diff
@@ -29,6 +29,8 @@ if [ ! -z ${AWS_CONTAINER_CREDENTIALS_RELATIVE_URI+x} ]; then
   echo "Running inside an ECS task, using container credentials"
 elif curl --output /dev/null --silent --head --fail --connect-timeout 2 "http://169.254.169.254"; then
   echo "Running inside an EC2 instance, using IMDS for credentials"
+elif [ ! -z ${AWS_WEB_IDENTITY_TOKEN_FILE+x} ]; then
```

Comment:
Can you add a comment above this line that explains the logic? If I don't know anything, I'll that I see is that if the `AWS_WEB_IDENTITY_TOKEN_FILE` environment variable exists, then we output a string. Please explain what is the significance of doing nothing.

Also, I believe you can do this and it is the same, so you don't need to do double negation. However, please test to make sure I'm not crazy.
```
elif [ -n ${AWS_WEB_IDENTITY_TOKEN_FILE+x} ]; then
```

Reply:
I added a few comments around this whole block of code. Here's a snipped to show the difference between the approaches (I added and used a third, more readable one):

```shell
#!/usr/bin/env bash

# a)
unset FOO
if [ ! -z ${FOO+x} ]; then
  echo "a) is true"
fi

# b)
FOO=""
if [ ! -z ${FOO+x} ]; then
  echo "b) is true"
fi

# c)
FOO="BAR"
if [ ! -z ${FOO+x} ]; then
  echo "c) is true"
fi

echo "###"

# d)
unset FOO
if [ -n ${FOO+x} ]; then
  echo "d) is true"
fi

# e)
FOO=""
if [ -n ${FOO+x} ]; then
  echo "e) is true"
fi

# f)
FOO="BAR"
if [ -n ${FOO+x} ]; then
  echo "f) is true"
fi

echo "###"

# g)
unset FOO
if [[ -v FOO ]]; then
  echo "g) is true"
fi

# h)
FOO=""
if [[ -v FOO ]]; then
  echo "h) is true"
fi

# i)
FOO="BAR"
if [[ -v FOO ]]; then
  echo "i) is true"
fi
```

Reply:
Wow! That snippet was informative. I learned something new. Thanks for sharing.